### DISCUSSION TITLE
hot-warm elasticsearch deployment assistance

### DISCUSSION THREAD

**Discussion by linsecnet** [Security-Onion-Solutions/securityonion#6138] [Unsupported Versions] [answered]
Hi Everyone,

I've been reading and searching for a viable answer to accommodate hot-warm elasticsearch deployment.

I have 2 search nodes and 1 manager node and I will like to achieve log retention for longer than 30 days.

I saw rollup-jobs within kibana stack management where you can configure a hot-warm-cold environment ( no documentation behind it)

I saw in version 2.3.8 you have so-elasticsearch-warm.yml in /opt/so/saltstack/default/salt/curator/files/action/ where you can configure a hot-warm deployment

I wanted to find out the best approach

1. Should I use ILM role-up jobs to configure hot-warm deployment? 
2. Or Should I use Curator to achieve the same thing?
3. Or should I use ILM and Curator combined? the ILM to achieve hot-warm and curator to manage the size.

Any guidance or ideas will be highly appreciated. 

Deployment is as followed:
1 - Manager Node
2- Search Nodes
4 - Forward nodes
1 - Fleet Node

**Comment by dougburks** [comment, marked as answer]
We use Curator rather than ILM:
https://docs.securityonion.net/en/2.3/curator.html#curator-vs-index-lifecycle-management-ilm

Hot-warm requires Elastic clustering rather than cross cluster search:
https://docs.securityonion.net/en/2.3/elasticsearch.html#elastic-clustering

**Reply by linsecnet** [reply]
Thank you @dougburks
